simpegEM/FDEM/FDEM.py

```diff
@@ -335,3 +438,277 @@ def calcFieldsDeriv(self, sol, freq, fieldType, v, adjoint=False):
+
+
+class ProblemFDEM_j(BaseFDEMProblem):
+    """
+        Using the H-J formulation of Maxwell's equations
+
+        .. math::
+            \\nabla \\times \\sigma^{-1} \\vec{J} + i\\omega\\mu\\vec{H} = 0
+            \\nabla \\times \\vec{H} - \\vec{J} = \\vec{J_s}
+
+        Since \(\\vec{J}\) is a flux and \(\\vec{H}\) is a field, we discretize \(\\vec{J}\) on faces and \(\\vec{H}\) on edges.
+
+        For this implementation, we solve for J using \( \\vec{H} = - (i\\omega\\mu)^{-1} \\nabla \\times \\sigma^{-1} \\vec{J} \) :
+
+        .. math::
+            \\nabla \\times ( \\mu^{-1} \\nabla \\times \\sigma^{-1} \\vec{J} ) + i\\omega \\vec{J} = - i\\omega\\vec{J_s}
+    
+        We discretize this to:
+
+        .. math::
+            (\\mathbf{C}  \\mathbf{M^e_{mu^{-1}}} \\mathbf{C^T} \\mathbf{M^f_{\\sigma^{-1}}}  + i\\omega ) \\mathbf{j} = - i\\omega \\mathbf{j_s}
+
+        .. note::
+            This implementation does not yet work with full anisotropy!!
+
+    """
+
+    solType = 'j'
+    storeTheseFields = ['j','h']
+
+    def __init__(self, model, **kwargs):
+        BaseFDEMProblem.__init__(self, model, **kwargs)
+
+    def getA(self, freq):
+        """
+            Here, we form the operator \(\\mathbf{A}\) to solce 
+            .. math::
+                    \\mathbf{A} = \\mathbf{C}  \\mathbf{M^e_{mu^{-1}}} \\mathbf{C^T} \\mathbf{M^f_{\\sigma^{-1}}}  + i\\omega
+
+            :param float freq: Frequency
+            :rtype: scipy.sparse.csr_matrix
+            :return: A
+        """
+
+        MeMuI = self.MeMuI
+        MfSigi = self.MfSigmai
+        C = self.mesh.edgeCurl
+        iomega = 1j * omega(freq) * sp.eye(self.mesh.nF)
+
+        return C * MeMuI * C.T * MfSigi + iomega
+
+
+    def getADeriv(self, freq, u, v, adjoint=False):
+        """
+            In this case, we assume that electrical conductivity, \(\\sigma\) is the physical property of interest (i.e. \(\sigma\) = model.transform). Then we want
+            .. math:: 
+                \\frac{\mathbf{A(\\sigma)} \mathbf{v}}{d \\mathbf{m}} &= \\mathbf{C}  \\mathbf{M^e_{mu^{-1}}} \\mathbf{C^T} \\frac{d \\mathbf{M^f_{\\sigma^{-1}}}}{d \\mathbf{m}}
+                &= \\mathbf{C}  \\mathbf{M^e_{mu}^{-1}} \\mathbf{C^T} \\frac{d \\mathbf{M^f_{\\sigma^{-1}}}}{d \\mathbf{\\sigma^{-1}}} \\frac{d \\mathbf{\\sigma^{-1}}}{d \\mathbf{\\sigma}} \\frac{d \\mathbf{\\sigma}}{d \\mathbf{m}}
+        """
+
+        MeMuI = self.MeMuI
+        C = self.mesh.edgeCurl
+        sig = self.curModel.transform
+        sigi = 1/sig
+        dsig_dm = self.curModel.transformDeriv
+        dsigi_dsig = -Utils.sdiag(sigi)**2
+        dMf_dsigi = self.mesh.getFaceInnerProductDeriv(sigi)(u)
+
+        if adjoint:
+            return dsig_dm.T * ( dsigi_dsig.T *( dMf_dsigi.T * ( C * ( MeMuI.T * ( C.T * v ) ) ) ) )
+
+        return C * ( MeMuI * ( C.T * ( dMf_dsigi * ( dsigi_dsig * ( dsig_dm * v ) ) ) ) ) 
+
+    # def getjs(self,freq):
```

Comment:
Can we kill commented code if it is not in use/will not be in use? I am not sure if that is the case here or not?